libraries/ESP8266WiFiMesh/src/Crypto.cpp

```diff
@@ -0,0 +1,1002 @@
+/**
```

Comment:
I've replaced the previous crypto lib with a BearSSL frontend named CryptoInterface. If you think that looks OK, it should be fairly trivial to just expose more of the BearSSL backend and make that into a full crypto library. I could expose all BearSSL HMAC and hash types, for sure. Perhaps also the AEAD encryption functionality. One worry I have though is that novice users will resort to MD5 hashing because it is the most well known hash type. As you probably know, MD5 is really broken in terms of attacker resistance, and the same goes for SHA1 these days. The question then is if we should expose these hash types with a strong warning or not expose them at all?

Reply:
I've updated the CryptoInterface once more, exposing all BearSSL HMAC and hash functionality. MD5 and SHA1 hash usage will result in a Serial print warning that can be disabled. Should I move the CryptoInterface code to a separate PR?

Reply:
Splitting into smaller PRs for individual consideration is always preferred :)

Reply:
PR #6961 created for the new crypto library.